File: LiMi/Misc/Libs/阿里云短视频/录制编辑/qusdk/Record/View/ShotCountDownView.m
//
//  ShotCountDownView.m
//  LiMi
//
//  Created by dev.liufeng on 2018/5/18.
//  Copyright © 2018年 Chengdu you Hong Science & Technology Co., Ltd. All rights reserved.
//

#import "ShotCountDownView.h"
@interface ShotCountDownView()
    @property (nonatomic,weak) UILabel *timeLabel;
    @property (nonatomic,strong) NSTimer *timer;
    @property (nonatomic,assign) int time;
@end
@implementation ShotCountDownView
    - (instancetype)init{
        if (self = [self initWithFrame:CGRectMake(0, 0, ScreenWidth, ScreenHeight)]){
            
        }
        return self;
    }
    
    - (instancetype)initWithFrame:(CGRect)frame{
        if (self=[super initWithFrame:frame]){
            CGRect timeLabelFrame = CGRectMake(0, 0, 200, 200);
            UILabel *timeLabel = [[UILabel alloc] initWithFrame:timeLabelFrame];
            self.timeLabel = timeLabel;
//            [self.timeLabel sizeToFit];
            self.timeLabel.center = self.center;
            self.timeLabel.textAlignment = NSTextAlignmentCenter;
            self.timeLabel.font = [UIFont systemFontOfSize:150 weight:UIFontWeightBold];
            self.timeLabel.textColor = UIColor.whiteColor;
            self.timeLabel.backgroundColor = UIColor.clearColor;
            [self addSubview:self.timeLabel];
            self.backgroundColor = rgba(0, 0, 0, 0.4);
        }
        return self;
    }
    
- (void)showWith:(int)time{
    self.time = time;
    [UIApplication.sharedApplication.keyWindow addSubview:self];
    self.timeLabel.text = [NSString stringWithFormat:@"%d",self.time];
    //[self.timeLabel sizeToFit];
    self.timer = [NSTimer scheduledTimerWithTimeInterval:1 repeats:true block:^(NSTimer * _Nonnull timer) {
        self.time -= 1;
        if(self.time <= 0){
                if(self.completeBlock != nil){
                    self.completeBlock();
                }
                [self.timer invalidate];
                 [self removeFromSuperview];
            
             }else{
                 self.timeLabel.text = [NSString stringWithFormat:@"%d",self.time];
             }
    }];
}


@end

File: LiMi/发布/View/ShotCountDownView.swift
//
//  ShotCountDownView.swift
//  LiMi
//
//  Created by dev.liufeng on 2018/5/17.
//  Copyright © 2018年 Chengdu you Hong Science & Technology Co., Ltd. All rights reserved.
//

import UIKit

class ShotCountDownView: UIView {
    var timeLabel:UILabel!
    var timer:Timer!
    var time:Int = 0
    var completeBlock:(()->Void)?
    convenience init() {
        self.init(frame: SCREEN_RECT)
    }
    override init(frame: CGRect) {
        super.init(frame: frame)
        self.timeLabel = UILabel.init()
        self.timeLabel.font = UIFont.systemFont(ofSize: 150, weight: .bold)
        self.timeLabel.textColor = UIColor.white
        self.addSubview(self.timeLabel)
        self.timeLabel.snp.makeConstraints {[unowned self] (make) in
            make.center.equalTo(self)
        }
        self.backgroundColor = RGBA(r: 0, g: 0, b: 0, a: 0.4)
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    func showWith(time:Int){
        self.time = time
        UIApplication.shared.keyWindow?.addSubview(self)
        self.timeLabel.text = "\(self.time)"
        if #available(iOS 10.0, *) {
            self.timer = Timer.scheduledTimer(withTimeInterval: 1, repeats: true, block: {[unowned self] (_) in
                self.time -= 1
                if self.time <= 0{
                    self.timer.invalidate()
                    self.removeFromSuperview()
                }else{
                    self.timeLabel.text = "\(self.time)"
                }
            })
        } else {
            // Fallback on earlier versions
        }
    }
}
